### DISCUSSION TITLE
Custom forms using RJSF type form definition

### DISCUSSION THREAD

**Discussion by gd-rvemula** [refinedev/refine#3995] [Integration request] [answered]
I am looking to see if custom forms can be defined with something like RJSF which uses a JSON to define the form structure. 

Any plans to integrate RTSF forms as custom forms? https://github.com/rjsf-team/react-jsonschema-form

It would be awesome feature to speed up custom form definition with something like RJSF.

**Comment by omeraplak** [comment, marked as answer]
Hey @gd-rvemula , since it's already built on a refine hook based structure, you can start using this library now. Maybe it would be nice to publish an example with this library. Does anyone want to contribute?

**Comment by gd-rvemula** [comment]
I created an example with Refine and RJSF, if anyone stumbles on this thread in future. 
https://github.com/gd-rvemula/RefineAntdInferencer-Rjsf

**Comment by gd-rvemula** [comment]
I could not get Refine REST dataprovider to work, welcome any advice to integrate Refine dataprovider.
